src/psyclone/tests/domain/lfric/lfric_loop_test.py

```diff
@@ -261,39 +260,23 @@ def test_lower_to_language_normal_loop():
     assert loop1.start_expr.symbol.name == "loop1_start"
 
 
-def test_lower_to_language_domain_loop():
-    ''' Tests that we can call lower_to_language_level on a domain LFRicLoop.
-    This test takes an invoke with two consecutive domain kernels and then
-    fuses the 'loops' to verify that the kernels are all still in the right
-    order.
+def test_lower_to_language_null_loop():
+    ''' Tests that we can call lower_to_language_level on a NULL LFRicLoop.
+    The NULL loop is replaced by a dummy loop with a single iteration
     '''
 
     _, invoke = get_invoke("25.1_kern_two_domain.f90", TEST_API, idx=0)
-    # Domain loops cannot be fused with the transformation, so manually
-    # move the two kernels into one domain loop. First detach the second
+    # For this test we want to put 2 kernels inside the same null loop
+    # because we want to check that the order of the statements is
+    # maintained after removing the loop.
+    # Null loops cannot be fused with the transformation, so manually
+    # move the two kernels into one loop. First detach the second
     # LFRicLoop from the invoke, then detach the actual kernel. Lastly,
     # insert this second kernel into the domain loop body:
     sched = invoke.schedule
-    loop1 = sched.children[1].detach()
-    kern = loop1.loop_body.children[0].detach()
-    sched.children[0].loop_body.children.insert(1, kern)
-
-    # Check that the loops are in the expected order - the first kernel
-    # uses a and f1, the second b and f2:
-    assert sched.children[0].loop_body.children[0].args[0].name == "a"
-    assert sched.children[0].loop_body.children[0].args[1].name == "f1"
-    assert sched.children[0].loop_body.children[1].args[0].name == "b"
-    assert sched.children[0].loop_body.children[1].args[1].name == "f2"
-
-    # This call removes the loop and replaces it with the actual kernel
-    # call in case of a domain loop. It also adds the implicit arguments
-    # so the variable names have a different index in the lowered tree:
+    assert isinstance(sched.children[1], LFRicLoop)
     sched.lower_to_language_level()
-    assert isinstance(sched[0], Call)
-    assert sched.children[0].children[2].name == "a"
-    assert sched.children[0].children[3].name == "f1_data"
-    assert sched.children[1].children[2].name == "b"
-    assert sched.children[1].children[3].name == "f2_data"
+    assert not isinstance(sched.children[1], LFRicLoop)
```

Comment:
Should it just be a `Loop` now? Please add a check.

Reply:
Please could you also check that the final 'body' of the loop is as expected (just check the `debug_string()` perhaps?).

Reply:
Aha this loop was testing a domain loop with 2 statements inside. 

In fact this is the reason why I may had the dummy loop in my previous version, PSyIRGen only accepts one node. For now I raise and error for domain loops with multiple statements as they can not be fused anyway, and add a new test for this. But then everything is part of PSyIR the lowering can add multiple statements to its parent lowered schedule.